Source/DafnyCore/Resolver/PreType/PreTypeToType.cs

```diff
@@ -0,0 +1,174 @@
+//-----------------------------------------------------------------------------
```

Comment:
As for the rest of the code in PreType, some high-level comments would be welcome

Reply:
+1